### ISSUE TITLE
Error when parsing html attribute

### ISSUE THREAD

**Issue by dohvis** [wilsonzlin/minify-html#34]
I use `minify-html == 0.4.10`, and found 3 bugs

```python
# Bug 1.
minify('<a href="\"">') # -> SyntaxError

# Bug 2.
text = '<meta type="og:url" content="ablog://main?a=b&amp;param=1" />'
result = minify(text) # 
# Actual '<meta type=og:url content=ablog://main?a=b&ampparam=1>'
# Expected '<meta type=og:url content=ablog://main?a=b&param=1>'

# Bug 3, might be related with bug 2
minify(result) # -> SyntaxError: Expected attribute name. [Character 40]
```

**Comment by wilsonzlin** [comment]
The first input is not a bug, as that is invalid input (the last quote is not a valid attribute name and there is no closing tag). It's expected that an error is raised.

The second input is an extension of #33; I'll investigate incorporating the obscure spec rule into the minification consideration for attribute values.

The third input is due to the parser following the official spec, but the minifier outputs code that would nonetheless work in most browsers. I've relaxed the parser to accept unquoted attribute values that contain [illegal characters](https://html.spec.whatwg.org/multipage/syntax.html#unquoted).

**Comment by dohvis** [comment]
> The first input is not a bug, as that is invalid input (the last quote is not a valid attribute name and there is no closing tag). It's expected that an error is raised.

```
minify('<a href="\'">') # -> '<a href="\'">'
```
Then is it correct? I expected `<a href="'">...` (no `\`)
Also, how can I use. double quote in attribute name? For example, `<input value="I'm \"octocat\" !" />`

**Comment by wilsonzlin** [comment]
Backslash is not a special character in HTML. The language you're using minify-html from may have special behaviour for interpreting backslashes in string literals.

You cannot use double quote characters in an attribute name, it's not permitted per the spec: https://html.spec.whatwg.org/multipage/syntax.html#syntax-attribute-name. Additionally, `/>` doesn't mean anything in HTML; it doesn't self-close the tag.

**Comment by wilsonzlin** [comment]
The second input has been fixed as of `0.5.0`; ampersands will not be encoded in attribute values if their named entity doesn't end with `;` and is followed by `[a-zA-Z0-9=]`. Let me know if it works for you.
